From 9c47d38b15d39f11a5414aadd5b24307753447c5 Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Wed, 22 Jul 2015 21:51:49 -0400
Subject: [PATCH 2/2] Add validation to ping the default gateway

This patch adds a new function that runs on all nodes so that
the default gateway is ping tested early on during the
installation process (before any puppet is executed)

Change-Id: Ida125e79f3e549cae66be9dbd562197afa6400a5
---
 validation-scripts/all-nodes.sh | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/validation-scripts/all-nodes.sh b/validation-scripts/all-nodes.sh
index 44206c9..7d6d760 100644
--- a/validation-scripts/all-nodes.sh
+++ b/validation-scripts/all-nodes.sh
@@ -2,7 +2,7 @@
 
 # For each unique remote IP (specified via Heat) we check to
 # see if one of the locally configured networks matches and if so we
-# attempt a ping test on that networks remote IP.
+# attempt a ping test the remote network IP.
 function ping_controller_ips() {
   local REMOTE_IPS=$1
 
@@ -26,4 +26,16 @@ function ping_controller_ips() {
   done
 }
 
+function ping_default_gateway() {
+  DEFAULT_GW=$(ip r | grep default | cut -d " " -f 3)
+  echo -n "Trying to ping default gateway ${DEFAULT_GW}..."
+  if ! ping -c 1 $DEFAULT_GW &> /dev/null; then
+    echo "FAILURE"
+    echo "$DEFAULT_GW is not pingable."
+    exit 1
+  fi
+  echo "SUCCESS"
+}
+
 ping_controller_ips "$controller_ctlplane_ip_address $controller_external_ip_address $controller_internal_api_ip_address $controller_storage_ip_address $controller_storage_mgmt_ip_address $controller_tenant_ip_address"
+ping_default_gateway
-- 
1.8.3.1

